### PR TITLE
fix: tainting: Treat lambda bound variables as fresh

### PR DESCRIPTION
**Pull request by IagoAbal**
Closes PA-3298

test plan:
make test # new test

### CODE REVIEW

File: changelog.d/pa-3298.fixed [hunk 1]
```diff
@@ -0,0 +1,10 @@
+taint-mode: Semgrep will now treat lambdas' parameters as fresh, so a taint rule
+that to find double-delete's should no trigger on the code below:
```
Comment (amchiclet) [inline changelog.d/pa-3298.fixed]: nit:
so a taint rule that to find double-delete's should no trigger on ->
so a taint rule that finds double-delete's should not be triggered on

**Review by amchiclet — APPROVED** [review]
These edge cases are so hard to think about in advance. I guess the best way is still to fix issues as they come up.

File: changelog.d/pa-3298.fixed (hunk 1 above)
Comment (IagoAbal) [reply changelog.d/pa-3298.fixed]: Thanks for the correction!

**Comment by IagoAbal** [comment]
> These edge cases are so hard to think about in advance. I guess the best way is still to fix issues as they come up.

In some cases perhaps we should open a draft PR first, ask SR to test the feature on real code (or tell us what to test), and give us the OK.
